src/coreclr/src/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -449,6 +452,17 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                 }
                 break;
 
+            // mvni won't take 8B/16B, so hard-code the size and reinterpret the bits. The behaviour is the same.
```

Comment:
In what way is the size being reinterpreted here?

Reply:
As in, we'll always use instructions for 32-bit sized elements [because MVNI doesn't allow 8/16 1-byte elements.](https://developer.arm.com/docs/ddi0596/e/simd-and-floating-point-instructions-alphabetic-order/mvni-move-inverted-immediate-vector). Behaviour-wise they are the same (because we invert no bits or all the bits).

Reply:
Ah, OK - might be better worded as "mvni doesn't support the range of element types, so hard code the 'opts' value." - I was confused into thinking that it was the `emitSize` that it was talking about.